osu.Game/Overlays/BeatmapListing/BeatmapListingResultsDisplay.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Overlays.Direct;
+using osuTK;
+
+namespace osu.Game.Overlays.BeatmapListing
+{
+    public class BeatmapListingResultsDisplay : Container
+    {
+        private NotFoundDrawable noResultsDrawable;
+        private FillFlowContainer<DirectPanel> resultPanels;
+
+        public BeatmapListingResultsDisplay()
+        {
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            noResultsDrawable = new NotFoundDrawable();
+            resultPanels = new FillFlowContainer<DirectPanel>()
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Spacing = new Vector2(10),
+                Alpha = 0,
+                Margin = new MarginPadding { Vertical = 15 }
+            };
+        }
+
+        public void ReplaceBeatmaps(List<BeatmapSetInfo> beatmaps)
+        {
+            clearBeatmapPanels();
+
+            if (beatmaps.Count > 0)
+            {
+                if (noResultsDrawable.IsAlive)
+                    noResultsDrawable.FadeOut(100, Easing.OutQuint).Then().Schedule(() => RemoveInternal(noResultsDrawable));
```

Comment:
I see no reason to keep re-instantiating the no results drawable as it's static content. Can be just faded in and out whenever the need is (would probably be less likely to bug out, too).